app/client/src/widgets/TableWidgetV2/component/cellComponents/DateCell.tsx

```diff
@@ -241,8 +250,9 @@ export const DateCell = (props: DateComponentProps) => {
   };
 
   const onPopoverClosed = () => {
+    const isValidValue = value !== undefined && value !== null;
```

Comment:
```suggestion
    const isValidValue = !!value;
```
Also Don't we have to consider `isRequired` here?

Reply:
we have to considering `isRequired` also

Reply:
Hi **@rahulbarwal** , will look into this and update the pr with expected results.